File: python code/basic functions/rotate.py
import cv2

# read image
source = cv2.imread("../../images/lena.jpg", flags=1)

# get the dimensions of the image
dim = source.shape

rotationAngle = 180
scaleFactor = 1

# rotate the image by 90 degrees about the center
# dim[0] stores the number of rows and dim[1] stores the number of columns
rotationMatrix = cv2.getRotationMatrix2D((dim[1] / 2, dim[0] / 2), rotationAngle, scaleFactor)

result = cv2.warpAffine(source, rotationMatrix, (dim[1], dim[0]))

# show images
cv2.imshow("Original", source)
cv2.imshow("Rotated Image", result)

cv2.waitKey(0)

File: c++ code/week1/examples/basic functions/basic functions/rotate.cpp
#include <opencv2/opencv.hpp>

int main3() {
	cv::Mat source, M, result;
	// read image
	source = cv::imread("../../../../../images/lena.jpg", cv::IMREAD_COLOR);

	cv::Point2f center((source.cols / 2.0), (1.0 * source.rows / 2.0));
	double rotationAngle = 30;
	double scale = 1;

	// getting the matrix which will define the rotation
	M = cv::getRotationMatrix2D(center, rotationAngle, scale);

	// rotate the source and store in result
	cv::warpAffine(source, result, M, cv::Size(source.cols, source.rows));

	// create windows to display
	cv::namedWindow("Original Image", cv::WINDOW_AUTOSIZE);
	cv::namedWindow("Rotated Image", cv::WINDOW_AUTOSIZE);

	// display images
	cv::imshow("Original Image", source);
	cv::imshow("Rotated Image", result);

	cv::waitKey(0);

	return EXIT_SUCCESS;
}